### PR TITLE
#154921351 chore:add locations service file

### PR DESCRIPTION
**Pull request by Mercy-Muchai**
#### What does this PR do?
locations service
#### Description of Task to be completed?
- create locations services to reuse in controllers
- refactor incidents & locations controllers
#### What are the relevant pivotal tracker stories?
[154921351](https://www.pivotaltracker.com/story/show/154921351)

### CODE REVIEW

File: server/controllers/incidents.js [hunk 1]
```diff
@@ -3,53 +3,46 @@ const User = require("../models").Users;
...
+        Incident
+        .create({
+          description: req.body.description,
+          subject: req.body.subject,
+          dateOccurred: req.body.dateOccurred,
+          userId: req.body.userId,
+          statusId: req.body.statusId || 1,
+          locationId,
+          categoryId: req.body.categoryId,
+          levelId: req.body.levelId,
```
Comment (Mnickii) [inline server/controllers/incidents.js]: remove this as this info isn't available to normal users.

File: server/controllers/incidents.js [hunk 2]
```diff
@@ -3,53 +3,46 @@ const User = require("../models").Users;
...
+      locationPromise.then(locationId => {
+        Incident
+        .create({
+          description: req.body.description,
+          subject: req.body.subject,
+          dateOccurred: req.body.dateOccurred,
+          userId: req.body.userId,
+          statusId: req.body.statusId || 1,
+          locationId,
+          categoryId: req.body.categoryId,
```
Comment (Mnickii) [inline server/controllers/incidents.js]: this too

Comment (Mnickii) [reply server/controllers/incidents.js]: the user can just enter the subject. We need to minimize necessary input for better User Experience.

Comment (Mnickii) [reply server/controllers/incidents.js]: @bmuthoga can these changes also be reflected in the bot?

File: server/controllers/incidents.js (hunk 1 above)
Comment (Mercy-Muchai) [reply server/controllers/incidents.js]: On the bot side, they don't input category. Level is optional to report a flag. They are both `allowNull: true`. Will retain them in the PUT for incidents for the admin to update those fields.

File: server/controllers/incidents.js (hunk 2 above)
Comment (Mercy-Muchai) [reply server/controllers/incidents.js]: Updated. @Mnickii

**Comment by Mnickii** [comment]
LGTM

**Pull request merged**